### PR TITLE
Fix reading packet on MQTT Dummy server

### PR DESCRIPTION
**Pull request by reprimande**
It didn't read every bytes when multiple packets are read on same time.
I found it on using load test on my local with using this dummy server.
Maybe it's improved. And the implementation isn't clean, but I think it's dummy and it needs to improve basic structure if it's for more clean...

### CODE REVIEW

File: v3/mqtt_dummy.go [hunk 1]
```diff
@@ -190,16 +191,26 @@ func handleSession(context *DummyContext, client *DummyClient) {
 
 		if err != nil {
 			logError("[MQTTD] Error reading packet from stream: %s", err)
-			keepConn = false
+			//keepConn = false
```
Comment (shigmas) [inline v3/mqtt_dummy.go]: Can we continue in this error case? Actually, I guess the client will cause this error, so I guess it's kind of a made up error?

File: v3/mqtt_dummy.go [hunk 2]
```diff
@@ -264,6 +271,26 @@ func getResponsePacket(client *DummyClient, pktBytes []byte,
 	}
 
 	return pkt, keepConn
+
+}
+
+func getResponsePacket(client *DummyClient, pktBytes []byte) ([]packet.Packet, []byte, bool) {
+	keepConn := true
```
Comment (shigmas) [inline v3/mqtt_dummy.go:281]: This is never changed, so we always return true.

**Review by shigmas — COMMENTED** [review]
Thanks for improving this. The dummy server was supposed to be simple, but grew to test lots of of edge cases.

My concern is about keepConn. I think the only time it will be false is on write error. But, I think the client can close the connection in another way?

File: v3/mqtt_dummy.go (hunk 1 above)
Comment (reprimande) [reply v3/mqtt_dummy.go]: Sorry, I forgot to remove it, it' for debug. But `break` is available on next line, so eventually this loop is finished on this case. Isn't it?

File: v3/mqtt_dummy.go (hunk 2 above)
Comment (reprimande) [reply v3/mqtt_dummy.go:281]: Sorry, I forgot to handle it. I've improve that.

**Review by shigmas — APPROVED** [review]
Cool. Thanks for fixing!

File: v3/mqtt_dummy.go (hunk 1 above)
Comment (shigmas) [reply v3/mqtt_dummy.go]: yep!
